GenderPayGap.WebJob/Functions/Functions.Purge.cs

```diff
@@ -27,48 +28,71 @@ public async Task PurgeUsers([TimerTrigger("40 3 * * *" /* 03:40 once per day */
             var runId = CreateRunId();
             var startTime = VirtualDateTime.Now;
             LogFunctionStart(runId,  nameof(PurgeUsers), startTime);
+            
+            DateTime deadline = VirtualDateTime.Now.AddDays(0 - Global.PurgeUnverifiedUserDays);
+            DateTime pinExpiryDate = VirtualDateTime.Now.AddDays(0 - Global.PinInPostExpiryDays);
+            
+            List<User> users = _DataRepository.GetAll<User>()
+                .Where(u => u.EmailVerifiedDate == null)
+                .Where(u => (u.EmailVerifySendDate == null || u.EmailVerifySendDate.Value < deadline))
+                .Include(u => u.UserOrganisations)
+                .ToList();
+            
+            foreach (User user in users)
+            {
+                PurgeUser(user, pinExpiryDate, runId, startTime);
+            }
+
+            LogFunctionEnd(runId, nameof(PurgeUsers), startTime);
+        }
+
+        private static async void PurgeUser(User user, DateTime pinExpiryDate, string runId, DateTime startTime)
+        {
             try
             {
-                DateTime deadline = VirtualDateTime.Now.AddDays(0 - Global.PurgeUnverifiedUserDays);
-                List<User> users = await _DataRepository.GetAll<User>()
-                    .Where(u => u.EmailVerifiedDate == null && (u.EmailVerifySendDate == null || u.EmailVerifySendDate.Value < deadline))
-                    .ToListAsync();
-                DateTime pinExpireyDate = VirtualDateTime.Now.AddDays(0 - Global.PinInPostExpiryDays);
-                foreach (User user in users)
+                //Ignore if they have verified PIN
+                if (user.UserOrganisations.Any(
+                    uo => uo.PINConfirmedDate != null || (uo.PINSentDate != null && uo.PINSentDate < pinExpiryDate)))

```

Comment:
I've run a query against both prod and dev but can't find any results that satisfy this line after having passed line 35. But I am still a bit apprehensive about changing the logic here.

Reply:
On reflection, I've decided to remove this block. Having run a query on both DEV and PROD I didn't find any users currently in this state. We also know that a user can not add an organisation until after they have been verified so users are unable to get into this state, which means this block should be redundant.